File: javascript/8kyu_count_sheep.js
var countSheep = function (num){
    var result = "";
    for (let i = 1; i <= num; i++) { 
        result += i + " sheep...";
      }
      return result;
  }

/*   var countSheep = function (num){
    let str = "";
    for(let i = 1; i <= num; i++) { str+= `${i} sheep...`; }
    return str;
  } */
countSheep(3)

const zeroFuel = (distanceToPump, mpg, fuelLeft) => {
  return mpg * fuelLeft >= distanceToPump;
};

function factorial(n){
  var i = 0;
  var result = 1;
  while (i < n) {
    result *= (n-i)
    i++;
  }
  return result
}

factorial(7)//, 5040);


function solution(a) {
  let counter = 0;
  let newIndex = 0;
  function theFunction (oldIndex){
  counter++;
  newIndex = oldIndex + a[oldIndex];
  console.log(newIndex)
  if (typeof a[newIndex] === 'undefined'){
  return counter;
  }
  else {
      theFunction(newIndex);
  };
  };
  let theCounter = theFunction(a[0]);
  return theCounter;
}
console.log(solution([1, 1, 1, 1]))//, 4);

File: python/8kyu_count_sheep.py
def count_sheep(n):
    result = ""
    for i in range(1,n+1):
        result += str(i)+ " sheep..."
    return result

""" def count_sheep(n):
    return "".join("{} sheep...".format(i) for i in range(1, n+1)) """

count_sheep(3)#, "1 sheep...2 sheep...3 sheep...")


def zero_fuel(distance_to_pump, mpg, fuel_left):
    return mpg * fuel_left >= distance_to_pump

zero_fuel(50, 25, 2)#, True)
zero_fuel(100, 50, 1)#, False)


def ordered_count(inp):
    return [(r,inp.count(r)) for r in sorted(set(inp), key=inp.index)]

ordered_count('abracadabra')
#('abracadabra', [('a', 5), ('b', 2), ('r', 2), ('c', 1), ('d', 1)]),

def sequence_sum(begin_number, end_number, step):
    sum = 0;
    for i in range(begin_number,end_number+1,step):
        sum += i;
    return sum;

sequence_sum(2, 6, 2)